### ISSUE TITLE
ability to patch XRC/XR'ss publishConnectionDetailsTo / publishConnectionDetailsWithStoreConfig field

### ISSUE THREAD

**Issue by jaylevin** [crossplane/crossplane#4087]
Hi,

We have multiple tenants deploying claims into their respective tenant namespace (claim namespace). We want to be able to have a `StoreConfig.secrets.crossplane.io` resource for each tenant/claim namespace, so that each tenant's XRC/XR connection details are pushed to a separate vault path.

We were able to successfully do this using MR-level secrets by patching the `publishConnectionDetailsTo.configRef.name` field of the MR, however I am confused on how to accomplish this with XRC/XR-level connection secrets.

I am mainly confused, because:

- With MR-level connection secrets, we can patch the `StoreConfig` reference per claim-namespace. This allows us to have a separate vault mountPath/scope for each tenant. 
- With XRC/XR-level connection secrets, there's no way to patch the fields  (ie the `publishConnectionDetailsWithStoreConfig` on the `Composition` resource). 

I must obviously be missing something simple. How do we accomplish this?

TL/DR:
- I want to patch the claim's `publishConnectionDetailsTo`, and abstract it from the user's view (the person submitting the claim shouldn't be able to control which `StoreConfig` `publishConnectionDetailsTo.configRef.name` is pointing to. The value should be automatically set depending on the claim namespace. 

Thanks in advance.
